### PR TITLE
Updates for text linting: Java agent, agent, OpAMP

### PR DESCRIPTION
**Pull request by svrnm**
Codifies #1661, makes sure that "agent" is written lower case across the docs where applicable. Also adds a rule to detect OpAMP

### CODE REVIEW

**Review by theletterf — APPROVED** [review]

File: .textlintrc.json [hunk 1]
```diff
@@ -16,7 +16,10 @@
         "/<http://.*>/",
         "medium.com/opentelemetry",
         "github.com/open-telemetry",
+        "`javaagent`",
```
Comment (chalin) [inline .textlintrc.json:19]: Why the code-fence ticks @svrnm?

**Review by chalin — COMMENTED** [review]
Nice additions! See inlined question.

File: .textlintrc.json (hunk 1 above)
Comment (chalin) [reply .textlintrc.json:19]: To elaborate: I don't think that the text linter checks words inside code fences. Let me check ... .

**Review by Traviscal — COMMENTED** [review]
https://t.me/traviiscalls

**Pull request merged**